Tests/PortalSwiftTests/PortalMPC/PortalMpcSpy.swift

```diff
@@ -0,0 +1,209 @@
+
+    // Generate Solana Wallet method tracking
+    var generateSolanaWalletCallsCount: Int = 0
+    var generateSolanaWalletUsingProgressCallbackParam: ((PortalSwift.MpcStatus) -> Void)?
+
+    // Generate Solana Wallet and Backup Shares method tracking
+    var generateSolanaWalletAndBackupSharesCallsCount: Int = 0
+    var generateSolanaWalletAndBackupSharesBackupMethodParam: PortalSwift.BackupMethods?
+    var generateSolanaWalletAndBackupSharesUsingProgressCallbackParam: ((PortalSwift.MpcStatus) -> Void)?
+
+    // Register Backup Method tracking
+    var registerBackupMethodCallsCount: Int = 0
+    var registerBackupMethodParam: PortalSwift.BackupMethods?
+    var registerBackupMethodStorageParam: PortalSwift.PortalStorage?
+
+    // GDrive Configuration tracking
+    var setGDriveConfigurationCallsCount: Int = 0
+    var setGDriveConfigurationClientIdParam: String?
+    var setGDriveConfigurationFolderNameParam: String?
+
+    // GDrive View tracking
+    var setGDriveViewCallsCount: Int = 0
+    var setGDriveViewParam: UIViewController?
+
+    // Passkey Authentication Anchor tracking
+    var setPasskeyAuthenticationAnchorCallsCount: Int = 0
+    var setPasskeyAuthenticationAnchorParam: ASPresentationAnchor?
+
+    // Passkey Configuration tracking
+    var setPasskeyConfigurationCallsCount: Int = 0
+    var setPasskeyConfigurationRelyingPartyParam: String?
+    var setPasskeyConfigurationWebAuthnHostParam: String?
+
+    // Password setting tracking
+    var setPasswordCallsCount: Int = 0
+    var setPasswordParam: String?
+
+    // Backup method tracking
+    var backupWithCompletionCallsCount: Int = 0
+    var backupWithCompletionMethodParam: PortalSwift.BackupMethods.RawValue?
+    var backupWithCompletionBackupConfigsParam: PortalSwift.BackupConfigs?
+    var backupWithCompletionProgressCallbackParam: ((PortalSwift.MpcStatus) -> Void)?
+
+    // Generate with completion tracking
+    var generateWithCompletionCallsCount: Int = 0
+    var generateWithCompletionProgressCallbackParam: ((PortalSwift.MpcStatus) -> Void)?
+
+    // Eject Private Key tracking
+    var ejectPrivateKeyCallsCount: Int = 0
+    var ejectPrivateKeyCipherTextParam: String?
+    var ejectPrivateKeyMethodParam: PortalSwift.BackupMethods.RawValue?
+    var ejectPrivateKeyBackupConfigsParam: PortalSwift.BackupConfigs?
+    var ejectPrivateKeyOrgBackupShareParam: String?
+
+    // Recover with completion tracking
+    var recoverWithCompletionCallsCount: Int = 0
+    var recoverWithCompletionCipherTextParam: String?
+    var recoverWithCompletionMethodParam: PortalSwift.BackupMethods.RawValue?
+    var recoverWithCompletionBackupConfigsParam: PortalSwift.BackupConfigs?
+    var recoverWithCompletionProgressCallbackParam: ((PortalSwift.MpcStatus) -> Void)?
+
+    func backup(_ method: PortalSwift.BackupMethods, usingProgressCallback: ((PortalSwift.MpcStatus) -> Void)?) async throws -> PortalSwift.PortalMpcBackupResponse {
```

Comment:
how do these functions allow us to test the functionality of the actual functions they represent?

Reply:
Like we if use this spy will the test simply only show us that the spy is acting the way we expect and not that the actual code is acting the same way?

Reply:
is there perhaps a platform that will auto generate spys for you based on real functions?

Reply:
i think i understand now. if we need to change an argument in the protocol well need to update the spy as well then yea?